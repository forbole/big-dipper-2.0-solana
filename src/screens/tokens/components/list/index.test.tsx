import React from 'react';
import renderer from 'react-test-renderer';
import {
  MockTheme, wait,
} from '@tests/utils';
import List from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};
jest.mock('next-translate/useTranslation', () => () => mockI18n);
jest.mock('@components', () => ({
  Box: (props) => <div id="Box" {...props} />,
  NoData: (props) => <div id="NoData" {...props} />,
  Search: (props) => <div id="Search" {...props} />,
}));

jest.mock('./components', () => ({
  Mobile: (props) => <div id="Mobile" {...props} />,
  Desktop: (props) => <div id="Desktop" {...props} />,
  Total: (props) => <div id="Total" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: Proposals/List', () => {
  it('matches snapshot', async () => {
    let component;
    renderer.act(() => {
      component = renderer.create(
        <MockTheme>
          <List
            items={[
              {
                token: 'BTC',
                price: 1802,
                marketCap: 515912496,
                address: '9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E',
              },
            ]}
            isItemLoaded={() => true}
            itemCount={1}
            loadMoreItems={() => null}
            handleSearch={jest.fn()}
            handleSort={jest.fn()}
            sortDirection="desc"
            sortKey="address"
          />
        </MockTheme>,
      );
    });
    await wait();

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
