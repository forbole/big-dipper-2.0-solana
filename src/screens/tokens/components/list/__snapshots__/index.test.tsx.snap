// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Proposals/List matches snapshot 1`] = `
<div
  className="makeStyles-root"
  id="Box"
>
  <div
    className="makeStyles-topContent"
  >
    <div
      className="makeStyles-total"
      id="Total"
      total="1"
    />
  </div>
  <div
    className="makeStyles-list"
  >
    <div
      id="Mobile"
      isItemLoaded={[Function]}
      itemCount={1}
      items={
        Array [
          Object {
            "address": "9n4nbM75f5Ui33ZbPYXn59EwSgE8CGsHtAeTH5YFeJ9E",
            "change": 10,
            "holders": 2178994,
            "marketCap": 515912496,
            "price": 1802,
            "token": "BTC",
            "volume": 15902496558,
          },
        ]
      }
      loadMoreItems={[Function]}
    />
  </div>
</div>
`;
